service-registry/registry-zero-config/src/main/java/org/apache/servicecomb/zeroconfig/ZeroConfigRegistryApplicationListener.java

```diff
@@ -52,32 +53,36 @@
   private static final Logger LOGGER = LoggerFactory
       .getLogger(ZeroConfigRegistryApplicationListener.class);
 
-  private ApplicationContext applicationContext;
   private MulticastSocket multicastSocket;
 
   @Override
   public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
 
     // same mechanism as Local registry to enable the Zero Config registry
-    String flag = System.getProperty(ZeroConfigRegistryConstants.ZERO_CONFIG_REGISTRY_FLAG);
-    if (flag != null && flag.equals(ZeroConfigRegistryConstants.ZERO_CONFIG_REGISTRY_ENABLE_FLAG)) {
-      this.applicationContext = applicationContext;
-      BeanUtils.setContext(applicationContext);
-      ServerUtil.init();
-      ClientUtil.init();
-
-      try {
-        this.multicastSocket = new MulticastSocket();
-      } catch (IOException e) {
-        LOGGER.error("Failed create MulticastSocket object", e);
-      }
-
-      Function<ServiceRegistry, ServiceRegistryClient> registryClientConstructor =
-          serviceRegistry -> new ZeroConfigRegistryClientImpl(new ZeroConfigRegistryService(),
-              multicastSocket, new RestTemplate());
-
-      ServiceRegistryConfig.INSTANCE.setServiceRegistryClientConstructor(registryClientConstructor);
+    boolean enable = DynamicPropertyFactory.getInstance()
+        .getBooleanProperty("servicecomb.zeroconfig.enabled", false).get();
```

Comment:
I think default value to 'true' is better. Because when users use zeroconfig, they have to add pom dependency implicitly. It is falst when zeroconfig module is not included. I set 'local/service center' default to 'true'. see https://github.com/apache/servicecomb-java-chassis/blob/master/service-registry/registry-local/src/main/java/org/apache/servicecomb/localregistry/LocalRegistration.java

Reply:
I see. let me change default value to true